app/grandchallenge/participants/models.py

```diff
@@ -54,3 +56,88 @@ def delete(self, *args, **kwargs):
 
     class Meta:
         unique_together = (("challenge", "user"),)
+
+
+def string_type_schema():
+    return {"type": "string"}
+
+
+class RegistrationQuestion(UUIDModel):
+    challenge = models.ForeignKey(
+        Challenge,
+        on_delete=models.CASCADE,
+    )
+
+    question_text = models.TextField(blank=False)
+    question_help_text = models.TextField(blank=True)
+
+    schema = models.JSONField(
+        default=string_type_schema,
+        help_text="A JSON schema definition against which an answer is validated. See https://json-schema.org/."
+        "Only Draft 7, 6, 4 or 3 are supported.",
+        validators=[JSONSchemaValidator()],
+    )
+
+    required = models.BooleanField(default=True)
+
+    @property
+    def has_answers(self):
+        return RegistrationQuestionAnswer.objects.filter(
+            question=self
+        ).exists()
+
+    class Meta:
+        unique_together = (("question_text", "challenge"),)
+
+    def clean(self):
+        super().clean()
+
+        if (
+            type(self)
+            .objects.filter(
+                challenge=self.challenge,
+                question_text=self.question_text,
+            )
+            .exclude(pk=self.pk)
+            .exists()
+        ):
+            raise ValidationError(
```

Comment:
Isn't this anyway raised in `full_clean` by the `unique_together` constraint?

Reply:
Huh, good point. I remember going expecting that but my practical tests showed: 
![image](https://github.com/user-attachments/assets/a49221ee-16e5-472f-b4fc-bd2fa01f321e)

So I thought it was more of a 'you better not try this via shell' sort of thing.

Turns out that the form saving does not care about any uniqueness constraints for non-included fields. One way to address this, just tested, would be to add a hidden challenge field and have it hidden. Also cleans up the weird instance challenge setting. So I'll follow that route.

Reply:
Ah, yes, hidden challenge field is definitely the way to do it.